### PR TITLE
searchBar with search, sort, filter

### PR DESCRIPTION
**Pull request by jana-scheidemann**
- fuse.js is used to implement searchBar (search icon that opens search-Modal) and search function
- transfer sort, filter and search logic to searchBar

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -39,14 +42,37 @@ export default function HomePage({
...
+  const fuse = new Fuse(projects, {
+    keys: ["title", "description", "materials", "steps.desc"],
+    includeScore: true,
+    threshold: 0.3,
+    shouldSort: true,
+    ignoreLocation: true,
+    ignoreFieldNorm: true,
+  });
+  const results = fuse.search(query);
+  console.log("results: ", results);
```
Comment (onemanwenttomow) [inline pages/index.js:58]: Whilst i am a big fan of logging during development, I don't think we want a console log making it into production.

File: pages/index.js [hunk 2]
```diff
@@ -39,14 +42,37 @@ export default function HomePage({
...
+  }
+
   return (
     <>
       <StyledHeadline>DIY APP</StyledHeadline>
-      <Navigation onAddProject={onAddProject} />
+      <Navigation
+        onAddProject={onAddProject}
+        toogleSortModal={toogleSortModal}
+        toogleFilterModal={toogleFilterModal}
```
Comment (onemanwenttomow) [inline pages/index.js:72]: The function name seems to have a typo

i.e. `toogle` => `toggle`

File: pages/index.js [hunk 3]
```diff
@@ -39,14 +42,37 @@ export default function HomePage({
...
+    ignoreLocation: true,
+    ignoreFieldNorm: true,
+  });
+  const results = fuse.search(query);
+  console.log("results: ", results);
+  const searchResults = query ? results.map((result) => result.item) : projects;
+
+  function handleSearch(event) {
+    const { value } = event.target;
+    setQuery(value);
```
Comment (onemanwenttomow) [inline pages/index.js:63]: **Debouncing Search Input**: If the project list is large you might want to consider debouncing the search input to reduce the number of searches performed while the user is typing. 

No need to implement it for this project. I just wanted you to be aware of the concept if you hadnt come across it before. 

Here is a [nice visualisation of debouncing](https://kettanaito.com/blog/debounce-vs-throttle).

The idea is that we might not want to perform a search on every keypress. (if it is an expensive operation)

If a user searches for `woodworking`

we might not want to search on every part of the query. 
`w`
`wo`
`woo`
`wood`
`woodw`
etc especially if the user is a fast typer. 

just on the final search term of `woodworking` 

There are a few good articles on debouncing (e.g [here](https://www.dhiwise.com/post/ultimate-guide-to-implementing-react-debounce-effectively))

**Review by onemanwenttomow — APPROVED** [review]
nice job team. 🎉

A couple of really small things that i found.

I also added some notes around the search, but they don't need to be implemented.

File: pages/index.js (hunk 3 above)
Comment (jana-scheidemann) [reply pages/index.js:63]: Thanks. That's a very good point. 
I decided to leave this as an idea for later, if maybe we will have a more complex list.

**Pull request merged**